cmd/history.go

```diff
@@ -0,0 +1,113 @@
+	"github.com/pulumi/pulumi/pkg/diag/colors"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+)
+
+func newHistoryCmd() *cobra.Command {
+	var stack string
+	var cmd = &cobra.Command{
+		Use:        "history",
+		Aliases:    []string{"hist"},
+		SuggestFor: []string{"updates"},
+		Short:      "Update history for a stack",
+		Long: "Update history for a stack\n" +
+			"\n" +
+			"This command lists data about previous updates for a stack.",
+		Args: cmdutil.NoArgs,
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+			s, err := requireStack(stack, false /*offerNew */, opts, false /*setCurrent*/)
+			if err != nil {
+				return err
+			}
+			b := s.Backend()
+			updates, err := b.GetHistory(commandContext(), s.Ref())
+			if err != nil {
+				return errors.Wrap(err, "getting history")
+			}
+
+			displayUpdate(updates, opts)
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"Choose an stack other than the currently selected one")
+	return cmd
+}
+
+func displayUpdate(updates []backend.UpdateInfo, opts display.Options) {
+	if len(updates) == 0 {
+		fmt.Println("Stack has never been updated")
+		return
+	}
+
+	printResourceChanges := func(backgroundColor string, textColor string, sign string, amountOfChange int, resetColor string) {
+		msg := opts.Color.Colorize(fmt.Sprintf("%s%s%s%v%s", backgroundColor, textColor, sign, amountOfChange, resetColor))
+		fmt.Print(msg)
+	}
+
+	for _, update := range updates {
+
+		fmt.Printf("UpdateKind: %v\n", update.Kind)
+		if update.Result == "succeeded" {
+			fmt.Print(opts.Color.Colorize(fmt.Sprintf("%sStatus: %v%s\n", colors.Green, update.Result, colors.Reset)))
+		} else {
+			fmt.Print(opts.Color.Colorize(fmt.Sprintf("%sStatus: %v%s\n", colors.Red, update.Result, colors.Reset)))
+		}
+		fmt.Printf("Message: %v\n", update.Message)
+
+		printResourceChanges(colors.GreenBackground, colors.Black, "+", update.ResourceChanges["create"], colors.Reset)
+		printResourceChanges(colors.RedBackground, colors.Black, "-", update.ResourceChanges["delete"], colors.Reset)
+		printResourceChanges(colors.YellowBackground, colors.Black, "~", update.ResourceChanges["update"], colors.Reset)
+		printResourceChanges(colors.BlueBackground, colors.Black, " ", update.ResourceChanges["same"], colors.Reset)
+
+		timeStart := time.Unix(update.StartTime, 0)
+		timeCreated := humanize.Time(timeStart)
+		timeEnd := time.Unix(update.EndTime, 0)
+		duration := timeEnd.Sub(timeStart)
+		fmt.Printf("%2sUpdated %s took %s\n", "", timeCreated, duration)
+
+		empty := func(s string) bool {
+			return len(strings.TrimSpace(s)) == 0
+		}
+
+		if len(update.Environment) != 0 {
+			indent := 4
+			if !empty(update.Environment["github.login"]) && !empty(update.Environment["git.committer"]) && !empty(update.Environment["git.committer.email"]) && !empty(update.Environment["git.head"]) {
```

Comment:
Instead of referencing these environment keys as string literals, consider using the enums defined in the `backend` package. That not only reduces the number of magic strings in the code, but also allows your editor to cross-like where the key is defined and where it is used. For example, so you can double check we are using the right keys.

https://github.com/pulumi/pulumi/blob/55bea6527690f24bb1dadfc10897212a99e84069/pkg/backend/updates.go#L64

For example, `github.login` appears to be used incorrectly here. It isn't the login of the _user_ who made the commit, but rather the GitHub organization/user that the code resides in. So perhaps `GitHub Repo: ${env[github.login]}/${env[github.repo]}` would be a more appropriate line to render?